### PR TITLE
feat: [PLG-2460]: CI Credits Sales led API implementation

### PR DESCRIPTION
**Pull request by Xin-Harness**
## Describe your changes
This PR provide harness support user access to grant Credits to customer.
Happy Path: Harness support user use its credentials to grant ci credentials.
Pasted Graphic 15
Pasted Graphic 17

Unhappy path: Customer can't direct access to the Credit grant.
Pasted Graphic 16
image

API is filtered out by Internal Auth filter by direct access
image

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49128)

### CODE REVIEW

**Review by thomas-grasman — DISMISSED** [review]
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49128)* status: 7 file(s) to review. 1 of 2 approvals obtained.

**Comment by Xin-Harness** [comment]
trigger uts

**Review by kapil-harness — APPROVED** [review]
> Reviewed 6 of 7 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49128)* status: 1 file(s) to review. 1 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/resources/AccountResource.java` line 711 at r2](https://reviewable.io/reviews/harness/harness-core/49128#-NXvC3Vx5Ypx5gGGitje:-NXvC3VyDpqdcTA2q2Xh:bi9qtnh) ([raw file](https://github.com/harness/harness-core/blob/d749bb281b0666a71f283addf44952fb2a2cce22/400-rest/src/main/java/software/wings/resources/AccountResource.java#L711)):*
> ```Java
> 
>   @POST
>   @Path("{accountId}/ng/credit")
> ```

nit: lets use accountIdentifier in place of accountId, in order to be consistent. Same change at other places in this class.

File: 930-ng-license-manager/src/main/java/io/harness/credit/resource/AdminCreditResource.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+  @Inject
+  public AdminCreditResource(CreditService creditService) {
+    this.creditService = creditService;
+  }
+
+  @POST
+  @Path("/create")
+  @ApiOperation(value = "Purchase credit for an account", nickname = "createCredit")
+  @Operation(operationId = "createCredit", summary = "Purchase the credits of an account",
```
Comment (neelam-harness) [inline 930-ng-license-manager/src/main/java/io/harness/credit/resource/AdminCreditResource.java:63]: Nit: `summary = Purchase credit for an account`

File: 930-ng-license-manager/src/main/java/io/harness/credit/resource/AdminCreditResource.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  createCredits(@Parameter(required = true, description = ACCOUNT_PARAM_MESSAGE) @NotNull @QueryParam(
+                    NGCommonEntityConstants.ACCOUNT_KEY) @AccountIdentifier String accountIdentifier,
+      @NotNull @Valid CreditDTO creditDTO) {
+    CreditDTO created = creditService.purchaseCredit(accountIdentifier, creditDTO);
+    return ResponseDTO.newResponse(created);
+  }
+
+  @GET
+  @Path("{accountIdentifier}")
+  @ApiOperation(value = "Get Credits purchase history of an Account in expiry time ascending order",
```
Comment (neelam-harness) [inline 930-ng-license-manager/src/main/java/io/harness/credit/resource/AdminCreditResource.java:80]: Nit: let's rephrase, `Get purchase history of credits for an Account in an ascending order of the expiry time`

**Review by neelam-harness — APPROVED** [review]
LGTM overall (very minor nit comments)

**Comment by Xin-Harness** [comment]
I will address some naming convention in a separate PR where I will also retire one direct access API

**Pull request merged**